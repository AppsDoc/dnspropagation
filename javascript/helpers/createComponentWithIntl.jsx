/**
 * @see https://github.com/yahoo/react-intl/wiki/Testing-with-React-Intl#jest
 */
import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';

const createComponentWithIntl = (children, props = { locale: 'en' }) =>
  renderer.create(
    <IntlProvider {...props}>
      {children}
    </IntlProvider>
  );

export default createComponentWithIntl;
