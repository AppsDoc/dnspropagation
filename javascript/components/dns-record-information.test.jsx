/*
 * The MIT License (MIT)
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
import React from 'react';
import DnsRecordInformation from './dns-record-information';
import createComponentWithIntl from '../helpers/createComponentWithIntl';

const params = {
  record: 'a',
  recordUpper: 'A'
};

test('it should render the component', () => {
  const tree = createComponentWithIntl(
    <DnsRecordInformation record={params.record} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('it should still render the component', () => {
  const tree = createComponentWithIntl(
    <DnsRecordInformation record={params.recordUpper} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
