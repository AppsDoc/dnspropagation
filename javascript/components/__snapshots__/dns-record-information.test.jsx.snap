// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render the component 1`] = `
<div
  className="block__dns-record-information"
>
  <blockquote>
    <p>
      <span>
        An A record maps a domain name to the IP address (IPv4) of the computer hosting the domain. Simply put, an A record is used to find the IPv4 address of a computer connected to the internet from a name. The A in A record stands for Address.
      </span>
    </p>
    <footer>
      https://support.dnsimple.com/articles/a-record
    </footer>
  </blockquote>
</div>
`;
